internal/services/network/network_manager_routing_rule_resource.go

```diff
@@ -0,0 +1,341 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/network/2024-05-01/routingrules"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var (
+	_ sdk.ResourceWithUpdate        = ManagerRoutingRuleResource{}
+	_ sdk.ResourceWithCustomizeDiff = ManagerRoutingRuleResource{}
+)
+
+type ManagerRoutingRuleResource struct{}
+
+func (ManagerRoutingRuleResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return routingrules.ValidateRuleID
+}
+
+func (ManagerRoutingRuleResource) ResourceType() string {
+	return "azurerm_network_manager_routing_rule"
+}
+
+func (ManagerRoutingRuleResource) ModelObject() interface{} {
+	return &ManagerRoutingRuleResourceModel{}
+}
+
+type ManagerRoutingRuleResourceModel struct {
+	Description      string                          `tfschema:"description"`
+	Destination      []ManagerRoutingRuleDestination `tfschema:"destination"`
+	Name             string                          `tfschema:"name"`
+	NextHop          []ManagerRoutingRuleNextHop     `tfschema:"next_hop"`
+	RuleCollectionId string                          `tfschema:"rule_collection_id"`
+}
+
+type ManagerRoutingRuleDestination struct {
+	Address string `tfschema:"address"`
+	Type    string `tfschema:"type"`
+}
+
+type ManagerRoutingRuleNextHop struct {
+	Address string `tfschema:"address"`
+	Type    string `tfschema:"type"`
+}
+
+func (ManagerRoutingRuleResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[a-zA-Z0-9\_\.\-]{1,64}$`),
+				"`name` must be between 1 and 64 characters long and can only contain letters, numbers, underscores(_), periods(.), and hyphens(-).",
+			),
+		},
+
+		"rule_collection_id": commonschema.ResourceIDReferenceRequiredForceNew(&routingrules.RuleCollectionId{}),
+
+		"destination": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"address": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
Shouldn't this be using the `validation.IsCIDR` and/or `validation.IsIPAddress` functions to verify this instead of just checking for and empty string here?

Reply:
This value can be a service tag if the destination type is `ServiceTag`. Since there are too many service tags, could we validate it as a non-empty string instead?
<img width="840" height="1264" alt="image" src="https://github.com/user-attachments/assets/3400667f-29a4-49f1-bcc0-3f56aaf47011" />